sql/binlog.cc

```diff
@@ -8789,7 +8789,11 @@ void MYSQL_BIN_LOG::process_after_commit_stage_queue(THD *thd, THD *first) {
       (void)RUN_HOOK(transaction, after_commit, (head, all));
 
       my_off_t pos;
-      head->get_trans_pos(nullptr, &pos, nullptr, nullptr);
+      const char *file_name;
+      head->get_trans_pos(&file_name, &pos, nullptr, nullptr);
+      // NO_LINT_DEBUG
+      sql_print_information("Signalling async threads to pos: %s:%llu",
```

Comment:
Doesn't this trigger printing to the error log after each group commit? We're only looking to print out the async threads position when a semi-sync slave connects to the server.

Reply:
Indeed, what I'm seeing in the error log for each insert statement on the master:

2020-01-10T02:21:00.973077Z 8 [Note] [MY-010462] [Repl] Start binlog_dump to master_thread_id(8) slave_server(2), pos(, 4)
2020-01-10T02:21:11.128362Z 9 [Note] [MY-000000] [Server] Signalling async threads to pos: master-bin.000001:374
2020-01-10T02:21:13.446666Z 9 [Note] [MY-000000] [Server] Signalling async threads to pos: master-bin.000001:648
2020-01-10T02:21:14.938622Z 9 [Note] [MY-000000] [Server] Signalling async threads to pos: master-bin.000001:922
2020-01-10T02:21:34.745018Z 9 [Note] [MY-000000] [Server] Signalling async threads to pos: master-bin.000001:1196
2020-01-10T02:21:36.329375Z 9 [Note] [MY-000000] [Server] Signalling async threads to pos: master-bin.000001:1470
2020-01-10T02:21:45.940968Z 9 [Note] [MY-000000] [Server] Signalling async threads to pos: master-bin.000001:1744